Index: /Users/jacob/Projects/Book/docutils-sandbox/OpenDocument/odtwriter/__init__.py
===================================================================
--- /Users/jacob/Projects/Book/docutils-sandbox/OpenDocument/odtwriter/__init__.py	(revision 5182)
+++ /Users/jacob/Projects/Book/docutils-sandbox/OpenDocument/odtwriter/__init__.py	(working copy)
@@ -817,6 +817,20 @@
 
     def depart_address(self, node):
         self.set_to_parent()
+        
+    def visit_admonition(self, node):
+        el = self.append_child('text:section', attrib={
+            'text:style-name': "rststyle-admonition",
+        })
+        self.set_current_element(el)
+        self.paragraph_style_stack.append("rststyle-admonition-textbody")
+        
+    def depart_admonition(self, node):
+        self.paragraph_style_stack.pop()
+        self.set_to_parent()
+        
+    visit_note = visit_admonition
+    depart_note = depart_admonition
 
     def visit_attribution(self, node):
         #ipshell('At visit_attribution')
@@ -1533,7 +1547,7 @@
 
     visit_doctest_block = visit_literal_block
     depart_doctest_block = depart_literal_block
-
+    
     def show_message(self, msg):
         #print '*** tagname: [[%s]] msg: %s' % (msg.starttag(), msg.astext(), )
         #print '*** msg: %s' % (dir(msg), )
@@ -1977,6 +1991,13 @@
             #text = text.decode('latin-1').encode('utf-8')
             el1.text = text
             self.title = text
+        elif isinstance(node.parent, docutils.nodes.admonition):
+            el1 = SubElement(self.current_element, "text:h", attrib = {
+                'text:style-name': 'rststyle-admonition-title'
+            })
+            text = node.astext()
+            el1.text = text
+            self.title = text
 
     def depart_title(self, node):
         pass
